client.go

```diff
@@ -620,11 +621,6 @@ func (client *client) tryRefreshMetadata(topics []string, attemptsRemaining int)
 		return err
 	}
 
-	isConnected, _ := client.waitForSingleConnection()
```

Comment:
The advantage to this check is that we avoid the retry loop if we're not connected to any brokers. Since the retry loop is intended to wait out leader elections, should we consider keeping the check and calling `client.resurrectDeadBrokers()` before return? The impact seems low either way, but keeping the check would cause less logspam.